internal/api/impl/auth/auth.go

```diff
@@ -141,3 +143,41 @@ func (e *endpoint) logout(ctx echo.Context) error {
 
 	return ctx.Redirect(302, "/")
 }
+
+// withOAuthErrorMdw applies to the handler a middleware that the error is well managed in oauth2.0 context.
+// This will transform any oauth2.0 retrieve token error into a bad request error, as suggested in the specs:
+// https://www.rfc-editor.org/rfc/rfc6749#section-5.2
+// https://www.rfc-editor.org/rfc/rfc8628#section-3.5
+// Otherwise, and only if the error is not an explicit Perses or HTTP error, we send back an opaque InternalError.
+func withOAuthErrorMdw(handler echo.HandlerFunc) echo.HandlerFunc {
+	return func(ctx echo.Context) error {
+		err := handler(ctx)
+		if err == nil {
+			return nil
+		}
+
+		urlError := &url.Error{}
+		if errors.As(err, &urlError) {
+			return urlError
```

Comment:
this error is not managed by Perses, so it will produce at the end an internalError.

What's the point of this check ?

Reply:
yes I forgot it, it was a check. I'm removing it